test/hotspot/jtreg/compiler/c2/cr7200264/TestIntVect.java

```diff
@@ -21,622 +21,763 @@
-      }
-
-    }
-
-    return errn;
-  }
-
-  static int test_sum(int[] a1) {
-    int sum = 0;
-    for (int i = 0; i < a1.length; i+=1) {
-      sum += a1[i];
-    }
-    return sum;
-  }
-
-  static void test_addc(int[] a0, int[] a1) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]+VALUE);
-    }
-  }
-  static void test_addv(int[] a0, int[] a1, int b) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]+b);
-    }
-  }
-  static void test_adda(int[] a0, int[] a1, int[] a2) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]+a2[i]);
-    }
-  }
-
-  static void test_subc(int[] a0, int[] a1) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]-VALUE);
-    }
-  }
-  static void test_subv(int[] a0, int[] a1, int b) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]-b);
-    }
-  }
-  static void test_suba(int[] a0, int[] a1, int[] a2) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]-a2[i]);
-    }
-  }
-
-  static void test_mulc(int[] a0, int[] a1) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]*VALUE);
-    }
-  }
-  static void test_mulc_n(int[] a0, int[] a1) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]*(-VALUE));
-    }
-  }
-  static void test_mulv(int[] a0, int[] a1, int b) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]*b);
-    }
-  }
-  static void test_mula(int[] a0, int[] a1, int[] a2) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]*a2[i]);
-    }
-  }
-
-  static void test_divc(int[] a0, int[] a1) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]/VALUE);
-    }
-  }
-  static void test_divc_n(int[] a0, int[] a1) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]/(-VALUE));
-    }
-  }
-  static void test_divv(int[] a0, int[] a1, int b) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]/b);
-    }
-  }
-  static void test_diva(int[] a0, int[] a1, int[] a2) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]/a2[i]);
-    }
-  }
-
-  static void test_andc(int[] a0, int[] a1) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]&BIT_MASK);
-    }
-  }
-  static void test_andv(int[] a0, int[] a1, int b) {
-    for (int i = 0; i < a0.length; i+=1) {
-      a0[i] = (int)(a1[i]&b);
+
+    private static final int ARRLEN = 997;
+    private static final int ITERS  = 11000;
+    private static final int ADD_INIT = Integer.MAX_VALUE-500;
+    private static final int BIT_MASK = 0xEC80F731;
+    private static final int VALUE = 15;
+    private static final int SHIFT = 32;
+
+    public static void main(String[] args) {
+        TestFramework.runWithFlags("-XX:+IgnoreUnrecognizedVMOptions",
+                                   "-XX:StressLongCountedLoop=0");
```

Comment:
Yes, thanks. The previous comment that fell out is `// make sure int loops do not get converted to long`. I'll readd it.

Reply:
And is that really necessary?
I don't see the flag in the IR Framework whitelist:
`test/hotspot/jtreg/compiler/lib/ir_framework/TestFramework.java`

That would mean that if you do set the flag, then we would already disable the IR rules, but we could still run the tests, checking for correctness.

TLDR: I believe we should just be able to remove this flag completely.

Reply:
Isn't that whitelist only for flags passed down through JTREG? For example, when I add `"-XX:LoopUnrollLimit=0"` (not in the whitelist) to the `runWithFlags` arguments, all the IR tests fail.

Reply:
Ok then, let me explain IR Framework whitelists, and `runWithFlags`:
Generally, we want a test to be run from the outside with as many flag combinations as possible, since some bugs only trigger with strange combinations.
On the other hand, we need some way to say under what conditions an IR rule should be checked, because many flags might have side-effects on the IR, and disable all sorts of optimizations.
We chose the whitelist approach: those flags are expected generally not to change the IR, or change it in a way that could also happen on other machines, and therefore must be allowed to simulate those machines (e.g. UseAVX).
If you have a test that is not ok with any combination of the whitelisted flags, then the test must further restrict the IR rule with `applyIf` statements.

Sometimes, we want to make sure that IR rules are ok with flag settings that are not allowed by the whitelist, for example `OptimizeFill` (it has an effect, but maybe an effect we want to check for in a specific test). Or maybe we want to overwrite some flag setting for specific reasons (put up some node limit, etc). In those cases, it can make sense to use `runWithFlags`.

In the case of this test, I don't see why you would want to set `-XX:LoopUnrollLimit=0` via `runWithFlags`. Of course this makes all tests fail, because it messes up unrolling, and without unrolling you have no SuperWord, and without SuperWord you have no vectorization.
But if anybody were to set this flag from the outside (via JTREG), then it would disable the IR rules implicitly (because the flag is not on the whitelist), and the jtreg test would pass as a whole.

Does that make sense, and help?

Reply:
Yes, thanks, that corresponds to my understanding of the framework. Sorry if I was unclear, we of course do not want to include `-XX:LoopUnrollLimit=0`, that was just an example to illustrate the difference between whitelisting and `runWithFlags` (and what originally spawned this issue: https://bugs.openjdk.org/browse/JDK-8291809).

The question is if we want to include `-XX:StressLongCountedLoop=0` in `runWithFlags`, as that flag was part of the original test. I'm fine with removing it if it no longer makes sense in the IR framework.

Reply:
Ok, great :)

I assume that `-XX:StressLongCountedLoop=0` was there before, because if it was not, then one could make the test fail with JTREG, by for example passing `-XX:StressLongCountedLoop=1000`, right?

But of course you would have to guard against all sorts of flags like this. For example `-Xint`. I just ran it with that flag, and it failed! :rofl:

Reply:
OK, I'll remove it then!